File: Domains/InfiniteMario/15-rl-competition-2009/system/codecs/Java/src/org/rlcommunity/rlglue/codec/tests/Test_Sanity_Experiment.java
/*
 * Copyright 2008 Brian Tanner
 * http://rl-glue-ext.ext.googlecode.com/
 * brian@tannerpages.com
 * http://brian.tannerpages.com
 * 
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package org.rlcommunity.rlglue.codec.tests;

import org.rlcommunity.rlglue.codec.RLGlue;

/**
 *
 * @author Brian Tanner
 */
public class Test_Sanity_Experiment {

    public static int runTest(){
        Glue_Test tester=new Glue_Test("Test_Sanity_Experiment");

        String task_spec=RLGlue.RL_init();
        
        tester.check_fail(!task_spec.equals("sample task spec"));

        
        System.out.println(tester);
        return tester.getFailCount();

    }
    public static void main(String[] args){
       System.exit(runTest());
    }

}

File: Domains/InfiniteMario/15-rl-competition-2009/system/codecs/Python/src/tests/test_sanity_experiment.py
# 
# Copyright (C) 2008, Brian Tanner
# 
#http://rl-glue-ext.googlecode.com/
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
#  $Revision: 617 $
#  $Date: 2009-02-05 04:24:12 -0500 (Thu, 05 Feb 2009) $
#  $Author: gabalz $
#  $HeadURL: http://rl-glue-ext.googlecode.com/svn/trunk/projects/codecs/Python/src/tests/test_sanity_experiment.py $

import sys

import rlglue.RLGlue as RLGlue
from glue_test import glue_test
tester =glue_test("test_sanity")

taskSpec=RLGlue.RL_init()
tester.check_fail(taskSpec!="sample task spec")

print tester.get_summary()
sys.exit(tester.getFailCount())
	
